File: saleup-ng/src/app/modules/sales/models/sale-request.ts
export class SaleRequest {
    saleId?: number;
    customerId?: number;
    saleItems?: SaleItemRequest[];
    subTotal?: number;
    discount?: number;
    deliveryAmount?: number;
    grandTotal?: number;
    notes?: string;
}
export class SaleItemRequest {
    saleItemId?: number;
    saleId?: number;
    productId?: number;
    quantity?: number;
}

File: saleup-spring/src/main/java/com/anaselrayan/springcashiero/features/sales/request/SaleRequest.java
package com.anaselrayan.springcashiero.features.sales.request;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.List;

@Data
public class SaleRequest {

    private Long saleId;

    private Long customerId;

    @NotNull(message = "sale item list can't be null")
    @Size(min = 1, message = "any sale can has at least one item")
    @Valid
    private List<SaleItemRequest> saleItems;

    @NotNull(message = "sub total can't be null")
    private Double subTotal;

    @NotNull(message = "grand total can't be null")
    private Double grandTotal;

    private Double discount;

}
